repo2docker/buildpacks/python/__init__.py

```diff
@@ -1,139 +1,66 @@
-        - support for JupyterLab
-        - support for nteract
-
-        """
-        return super().get_build_scripts() + [
-            (
-                "root",
-                r"""
-                mkdir -p ${VENV_PATH} && \
-                chown -R ${NB_USER}:${NB_USER} ${VENV_PATH}
-                """
-            ),
-            (
-                "${NB_USER}",
-                r"""
-                python3 -m venv ${VENV_PATH}
-                """
-            ),
-            (
-                "${NB_USER}",
-                r"""
-                pip install --no-cache-dir -r /tmp/requirements.frozen.txt && \
-                jupyter nbextension enable --py widgetsnbextension --sys-prefix && \
-                jupyter serverextension enable --py jupyterlab --sys-prefix && \
-                jupyter serverextension enable nteract_on_jupyter --sys-prefix
-                """
-            )
-        ]
+        try:
+            with open(self.binder_path('runtime.txt')) as f:
+                runtime = f.read().strip()
+        except FileNotFoundError:
+            runtime = 'python-' + self.major_pythons['3']
+        py_version_info = runtime.split('-', 1)[1].split('.')
+        py_version = ''
+        if len(py_version_info) == 1:
+            py_version = self.major_pythons[py_version_info[0]]
+        else:
+            # get major.minor
+            py_version = '.'.join(py_version_info[:2])
+        self._python_version = py_version
+        return self._python_version
 
     def get_assemble_scripts(self):
         """Return series of build-steps specific to this repository.
         """
         # If we have a runtime.txt & that's set to python-2.7,
         # we will *not* install requirements.txt but will find &
         # install a requirements3.txt file if it exists.
-        # This way, when using python2 venv, requirements.txt will
-        # be installed in the python2 venv, and requirements3.txt
-        # will be installed in python3 venv. This is less of a
+        # This way, when using python2 env, requirements.txt will
+        # be installed in the python2 env, and requirements3.txt
+        # will be installed in python3 env. This is less of a
         # surprise than requiring python2 to be requirements2.txt tho.
         assemble_scripts = super().get_assemble_scripts()
         setup_py = 'setup.py'
-        try:
-            with open(self.binder_path('runtime.txt')) as f:
-                runtime = f.read().strip()
-        except FileNotFoundError:
-            runtime = 'python-3.5'
-        if runtime == 'python-2.7':
-            pip = "pip2"
-            requirements_file = self.binder_path('requirements3.txt')
-        else:
-            pip = "pip3"
-            requirements_file = self.binder_path('requirements.txt')
+        pip = 'pip'
+        if self.py2:
+            # using python 2 kernel,
+            # requirements3.txt allows installation in the notebook env
```

Comment:
-> ... in the notebook server env. For example to install an additional notebook extension.

I had to read the comment a few times before I realised what it was good for. The 3 at the end of the filename was confusing me "why are we installing python 3 things with python 2 in use?????"